chainstate/src/detail/tests/double_spend_tests.rs

```diff
@@ -224,8 +227,55 @@ fn double_spend_tx_in_another_block() {
     });
 }
 
+// Try to process a block where the second transaction's input is more then first output.
+//
+// +--Block----------------+
+// |                       |
+// | +-------tx-1--------+ |
+// | |input = prev_block | |
+// | +-------------------+ |
+// |                       |
+// | +-------tx-2--------+ |
+// | |input = tx1        | |
+// | +-------------------+ |
+// +-----------------------+
+#[test]
+fn spend_bigger_output_in_the_same_block(atoms1 in 1000..2000u128, atoms2 in 100_000..200_000u128){
+    common::concurrency::model(move || {
+        let mut chainstate = setup_chainstate();
+
+        let first_tx = tx_from_genesis(&chainstate, atoms1);
+        let second_tx = tx_from_tx(&first_tx, atoms2);
+
+        let block = Block::new(
+            vec![first_tx, second_tx],
+            Some(Id::new(chainstate.chain_config.genesis_block_id().get())),
+            BlockTimestamp::from_duration_since_epoch(time::get()).unwrap(),
+            ConsensusData::None,
+        )
+        .expect(ERR_CREATE_BLOCK_FAIL);
+
+        assert_eq!(
+            chainstate.process_block(block, BlockSource::Local).unwrap_err(),
+            BlockError::StateUpdateFailed(StateUpdateError::AttemptToPrintMoney(
+                Amount::from_atoms(atoms1),
+                Amount::from_atoms(atoms2)
+            ))
+        );
+        assert_eq!(
+            chainstate
+                .chainstate_storage
+                .get_best_block_id()
+                .expect(ERR_BEST_BLOCK_NOT_FOUND)
+                .expect(ERR_STORAGE_FAIL),
+            chainstate.chain_config.genesis_block_id()
+        );
+    });
+}
+}
+
 // Creates a transaction with an input based on the first transaction from the genesis block.
-fn tx_from_genesis(chainstate: &Chainstate) -> Transaction {
+fn tx_from_genesis(chainstate: &Chainstate, atoms: u128) -> Transaction {
```

Comment:
I think that a better argument name would be something like `output_value`.

Reply:
To be clear, atoms is the equivalent of satoshis. As a name they're meaningless without context.